src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -144,11 +155,126 @@ static void create_static_function_call(
   }
 }
 
+/// If there are ASSERT or ASSUME instructions involving \p argument_for_this
+/// in the instructions directly preceding the virtual function call then we
+/// duplicate them for \p temp_var_for_this. This is mainly aimed at null
+/// checks, to make the local safe pointer analysis work. We stop as soon as we
```

Comment:
`to make the local safe pointer analysis work` -> `because \ref local_safe_pointerst would otherwise lose track of known-not-null pointers due to the newly introduced assignment`